src/k2/dto/SKVRecord.h

```diff
@@ -252,7 +257,8 @@ class SKVRecord {
 
     std::vector<String> partitionKeys;
     std::vector<String> rangeKeys;
-    uint32_t fieldCursor = 0;
+    uint32_t serializerFieldCursor = 0;    // field possition during serialization
+    uint32_t deserializerFieldCursor = 0;  // field possition during deserialization 
```

Comment:
There is a constructor for the SKVRecord that takes a Storage argument. In that constructor deserializeFieldCursor should be initialized to the end, since if the SKVRecord is constructed from an existing Storage then all the fields are already serialized.